core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -466,9 +466,37 @@ public void setUpdateTime(Date updatedAt) {
      * exist to allow for distributed contracts.
      */
     public enum SigHash {
-        ALL,         // 1
-        NONE,        // 2
-        SINGLE,      // 3
+        ALL(1),
+        NONE(2),
+        SINGLE(3),
+        ANYONECANPAY(0x80),
+        ANYONECANPAY_ALL(0x81),
+        ANYONECANPAY_NONE(0x82),
+        ANYONECANPAY_SINGLE(0x83),
+        UNSET(0);
```

Comment:
Do we need an UNSET value? In the Java world, this would typically be represented as `null`.